JWT/JWT.cs

```diff
@@ -103,34 +104,38 @@ public static string Decode(string token, byte[] key, bool verify = true)
                 var decodedCrypto = Convert.ToBase64String(crypto);
                 var decodedSignature = Convert.ToBase64String(signature);
 
-                if (decodedCrypto != decodedSignature)
-                {
-                    throw new SignatureVerificationException(string.Format("Invalid signature. Expected {0} got {1}", decodedCrypto, decodedSignature));
-                }
-
-                // verify exp claim https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-32#section-4.1.4
-                var payloadData = jsonSerializer.Deserialize<Dictionary<string, object>>(payloadJson);
-                if (payloadData.ContainsKey("exp") && payloadData["exp"] != null)
-                {
-                    // safely unpack a boxed int 
-                    int exp;
-                    try { exp = (int)payloadData["exp"]; }
-                    catch (Exception) 
-                    { 
-                        throw new SignatureVerificationException("Claim 'exp' must be an integer.");
-                    }
-
-                    var secondsSinceEpoch = Math.Round((DateTime.UtcNow - new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)).TotalSeconds);
-                    if (secondsSinceEpoch >= exp)
-                    {
-                        throw new SignatureVerificationException("Token has expired.");
-                    }
-                }
+                Verify(decodedCrypto, decodedSignature, payloadJson);
             }
 
             return payloadJson;
         }
 
+        private static void Verify(string decodedCrypto, string decodedSignature, string payloadJson)
+        {
+            if (decodedCrypto != decodedSignature)
+            {
+                throw new SignatureVerificationException(string.Format("Invalid signature. Expected {0} got {1}", decodedCrypto, decodedSignature));
+            }
+
+            // verify exp claim https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-32#section-4.1.4
+            var payloadData = jsonSerializer.Deserialize<Dictionary<string, object>>(payloadJson);
+            object value;
+            if (payloadData.TryGetValue("exp", out value) && value != null)
+            {
+                // safely unpack a boxed int
+                if (!(value is int))
```

Comment:
My tests show that it won't work either way, e.g.:
```csharp
int f = (int)(object)5f;
int d = (int)(object)5d;
int m = (int)(object)5m;
```
If you box long, uint, etc. and then try to unbox as int it will throw IvalidCastException.